### PR TITLE
Feat: QuickSort and sorting test suite

### PR DESCRIPTION
**Pull request by Xoeseko**
implemented quicksort not sure about whether I should have used filter or implemented a filtering myself. Wrote some quicktests to check some functionality that was buggy in my implementation namely repeating values and a random array for the general case.

### CODE REVIEW

**Comment by sentenza** [comment]
You should merge your branch with master and fix the conflicts. If you want to submit a better implementation of the current quicksort you may want to provide a brief explanation using this conversation. ;)

**Comment by Xoeseko** [comment]
Yes that's right I didn't realize quicksort was already implemented when I last pulled. I should have merged well. Here is what is left I hope my tests are useful anyway.

File: CONTRIBUTORS.md [hunk 1]
```diff
@@ -15,3 +17,4 @@
   - About: None
   - Programming Language: None
   - Website: https://flatmap.it
+>>>>>>>>> Temporary merge branch 2
```
Comment (sentenza) [inline CONTRIBUTORS.md]: You should remove this placeholder, that has been left behind by `git merge`.

File: .gitignore [hunk 2]
```diff
@@ -38,3 +38,5 @@ project/plugins/project/
 # System Files
 .DS_Store
 Thumbs.db
+
+*.sc
```
Comment (sentenza) [inline .gitignore]: Why did you add `*.sc` here?

File: src/main/scala/io/github/sentenza/hacktoberfest18/sort/CommonSortings.scala [hunk 3]
```diff
@@ -1,5 +1,7 @@
 package io.github.sentenza.hacktoberfest18.sort
 
+import java.util.Random
+
```
Comment (sentenza) [inline src/main/scala/io/github/sentenza/hacktoberfest18/sort/CommonSortings.scala]: We don't need this import statement any more, do we?

File: src/test/scala/io/github/sentenza/hacktoberfest18/sort/SortTests.scala [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+import util.Random
+import io.github.sentenza.hacktoberfest18.sort.CommonSortings.CommonSorting._
+
+/**
+  * This class is a test suite for the methods in object CommonSortings. To run
+  * the test suite, you can either:
+  *  - run the "test" command in the SBT console
+  *  - right-click the file in zour editor of choice and 'run as test' or 'run'
+  */
+class SortTests extends FunSuite {
```
Comment (sentenza) [inline src/test/scala/io/github/sentenza/hacktoberfest18/sort/SortTests.scala]: I'm afraid but I cannot accept a duplicate of what we have already under `test`. If you want to enhance what's already in there you should add more test cases.

File: .gitignore (hunk 2 above)
Comment (Xoeseko) [reply .gitignore]: I use worksheets for prototyping and thought it could be useful if others so to so that they are ignored as they are generally not implementation files.. I could have explicites that's true

File: src/main/scala/io/github/sentenza/hacktoberfest18/sort/CommonSortings.scala (hunk 3 above)
Comment (Xoeseko) [reply src/main/scala/io/github/sentenza/hacktoberfest18/sort/CommonSortings.scala]: No.

**Comment by Xoeseko** [comment]
In any case thanks for the feedback I'll work at the existing issues when I find some extra time this is all valuable experience

**Comment by sentenza** [comment]
> In any case thanks for the feedback I'll work at the existing issues when I find some extra time this is all valuable experience

You may want to start with the Bucket Sort algorithm #16. 👍
